OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -223,6 +223,8 @@ public IEnumerable<ProductionItem> AllQueued()
 
 		public virtual IEnumerable<ActorInfo> AllItems()
 		{
+			if (productionTraits.Any() && productionTraits.All(p => p.IsTraitDisabled))
```

Comment:
What happens in the case where some but not all `productionTraits` are disabled?  Does `allProducibles` return the correct result in this case?

Reply:
This method is used for the ClassicProductionQueue only, so it only cares about either all traits being disabled -> no production, or at last 1 trait enabled -> production available.